packages/babel-plugin-transform-optional-chaining/src/index.js

```diff
@@ -0,0 +1,117 @@
+import syntaxOptionalChaining from "babel-plugin-syntax-optional-chaining";
+
+export default function ({ types: t }) {
+  function optional(path, key, replacementPath, needsContext = false, loose = false) {
+    const { scope } = path;
+    const optionals = [path.node];
+    const nil = scope.buildUndefinedNode();
+
+    let objectPath = path.get(key);
+    while (objectPath.isMemberExpression()) {
+      const { node } = objectPath;
+      if (node.optional) {
+        optionals.push(node);
+      }
+
+      objectPath = objectPath.get("object");
+    }
+
+    for (let i = optionals.length - 1; i >= 0; i--) {
```

Comment:
Oh, right. `.reverse()` could do the trick, but the `i` would have to be subtracted from the `length` 🤔